### PR TITLE
vale-ls: init at 0.3.7-unstable-2024-03-13

### PR DESCRIPTION
**Pull request by jansol**
## Description of changes

Supersedes #288450 which seems to have stalled. This addresses most of the comment left in the old PR by @oliviacrain . I didn't find any decent way to pass the `installVale` init parameter -- it seems that has to be sent by the LSP client as part of the handshake process? Either way, although the documentation says it's enabled by default, the actual code seems to never initialize the field, meaning it is `None` which is treated as disabled?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by oliviacrain** [comment]
Will find time to review this later today, but at first glance, it looks like all my feedback has been addressed!

File: pkgs/by-name/va/vale-ls/package.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    pkg-config
+    makeWrapper
+  ];
+
+  buildInputs = [ openssl ];
+
+  # The following tests are reaching to the network.
+  checkFlags = [
+    "--skip=vale::tests"
+  ];
```
Comment (oliviacrain) [inline pkgs/by-name/va/vale-ls/package.nix:32]: Can you something along the following lines, so this package isn't broken on `aarch64`?

```suggestion
  checkFlags = [
    # The following tests are reaching to the network.
    "--skip=vale::tests"
  ] ++ lib.optionals (stdenv.isLinux && stdenv.isAarch64) [
    # This test does not account for the existence of aarch64-linux machines,
    # despite upstream shipping artifacts for that architecture
    "--skip=utils::tests::arch"
  ];
```

**Review by oliviacrain — APPROVED** [review]
Tested with [`helix`](https://github.com/helix-editor/helix) on `aarch64-linux`, appears to work for me! Thanks for picking this up!

**Comment by Aleksanaa** [comment]
```

Running phase: installPhase
Running phase: fixupPhase
shrinking RPATHs of ELF executables and libraries in /nix/store/gwq9a7lqlm8i190bhlszindvibjr5xwb-vale-ls-0.3.7-unstable-2024-03-13-vendor.tar.gz
checking for references to /build/ in /nix/store/gwq9a7lqlm8i190bhlszindvibjr5xwb-vale-ls-0.3.7-unstable-2024-03-13-vendor.tar.gz...
patching script interpreter paths in /nix/store/gwq9a7lqlm8i190bhlszindvibjr5xwb-vale-ls-0.3.7-unstable-2024-03-13-vendor.tar.gz
error: hash mismatch in fixed-output derivation '/nix/store/mixk6dkkv8013hv0lh8plbsbrz4lfl2f-vale-ls-0.3.7-unstable-2024-03-13-vendor.tar.gz.drv':
         specified: sha256-JDsveGMQaDLFOba6oKYdm14IYDUf4+FIG4dPm09zWog=
            got:    sha256-ifKdSTmVWfDZF5Kn9b5JpzDxa160oRTfzjvxeL9POBg=
error: 1 dependencies of derivation '/nix/store/ilaza77a8zxradx3hsdizfkpj4gxsflc-vale-ls-0.3.7-unstable-2024-03-13.drv' failed to build
```

@ofborg build vale-ls

**Comment by jansol** [comment]
Getting the same result here, I wonder if something got yanked & updated from crates.io? Or does the nix package version change also affect `cargoHash`?

**Comment by Aleksanaa** [comment]
> Getting the same result here, I wonder if something got yanked & updated from crates.io?

Maybe some upstream did something very unscrupulous like moving a tag.

> Or does the nix package version change also affect `cargoHash`?

That's unlikely to happen, don't worry

**Comment by jansol** [comment]
(Rebased on lastest master)

Okay this is interesting. When I build from my local copy I get the cargoHash specified in the derivation, but when I do `nix build github:jansol/nixpkgs/vale-ls#vale-ls` I get the one in your error message.

**Comment by jansol** [comment]
Nvm looks like the hash change was indeed caused by addressing the review comments (which I'd forgot to pull into my local branch). Now they match up as expected.

**Pull request merged**